src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -50,7 +50,39 @@ public override async Task OnConnectedAsync(HubConnectionContext connection)
                 try
                 {
                     InitializeHub(hub, connection);
-                    await hub.OnConnectedAsync();
+
+                    var hubFilters = ResolveHubFilters(connection, scope.ServiceProvider);
+                    if (hubFilters != null)
+                    {
+                        bool isConnected = false;
+                        Func<HubCallerContext, Task> app = async (context) =>
+                        {
+                            // Need to check if this was called, if it wasn't one of the filters is refusing the connection
+                            isConnected = true;
+                            try
+                            {
+                                await hub.OnConnectedAsync();
+                            }
+                            catch
+                            {
+                                // If OnConnectedAsync threw we want to close the connection, we can't trust Hub filters to not swallow the exception
+                                // so we need to check if there was one here
+                                isConnected = false;
```

Comment:
We should consider changing from `ValueTask<object>` to `ValueTask<HubResult>` where `HubResult` is a struct with a bool and object.

Reply:
Was this comment meant to be on ExecuteHubMethod? Regardless, I do like having a custom result type for discoverability if we ever add functionality.

Another option, that I like less, is to have InvokeMethodAsync return a plain ValueTask and make the return value a settable property on HubInvocationContext (or something derived from it).